Bolts/src/bolts/Task.java

```diff
@@ -223,6 +223,75 @@ public void run() {
   }
 
   /**
+   * Creates a task that will complete when any of the supplied tasks have completed.
+   * <p/>
+   * The returned task will complete when any of the supplied tasks has completed. The returned task
+   * will always end in the completed state with its result set to the first task to complete. This
+   * is true even if the first task to complete ended in the canceled or faulted state.
+   *
+   * @param tasks
+   *          The tasks to wait on for completion.
+   * @return A task that represents the completion of one of the supplied tasks. 
+   *         The return task's result is the task that completed.
+   */
+  public static <TResult> Task<Task<TResult>> whenAnyResult(Collection<Task<TResult>> tasks) {
+    if (tasks.size() == 0) {
+      return Task.forResult(null);
+    }
+
+    final Task<Task<TResult>>.TaskCompletionSource firstCompleted = Task.<Task<TResult>> create();
+    final AtomicBoolean isAnyTaskComplete = new AtomicBoolean(false);
+
+    for (final Task<TResult> inputTask : tasks) {
+        inputTask.continueWith(new Continuation<TResult, Void>() {
+        @Override
+        public Void then(Task<TResult> task) {
+          if (isAnyTaskComplete.compareAndSet(false, true)) {
+            firstCompleted.setResult(inputTask);
```

Comment:
you can replace `inputTask` with `task`, since `then` calls through with `this`. you can also remove the `final` modifier on `inputTask`.